### PR TITLE
Keep comment reply state in state instead of qs

### PR DESCRIPTION
**Pull request by ajacksified**
This moves whether or not a comment reply form is collapsed to a new key in the state object - replyingComments[id] - instead of in the querystring. Querystrings still work for no-js clients, but an onClick event captures and stops propagation for js-enabled clients.

👓 @schwers

### CODE REVIEW

**Comment by schwers** [comment]
Leaving a note from discussing in person; we don't actively support NoJS for posting so we should remove the query parameter entirely.

File: src/app/reducers/replyingComments.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import merge from '@r/platform/merge';
+
+import * as commentActions from 'app/actions/comment';
+import * as loginActions from 'app/actions/login';
+
+const DEFAULT = {};
```
Comment (schwers) [inline src/app/reducers/replyingComments.js]: One pattern we found helpful is to export the default state, then your tests don't have to be aware of what exactly the default is. Could you give that a try here?

File: src/app/pages/Comments.jsx [hunk 2]
```diff
@@ -60,15 +62,19 @@ const stateProps = createSelector(
   },
 );
 
-const dispatchProps = dispatch => ({
+const dispatchProps = (dispatch) => ({
```
Comment (schwers) [inline src/app/pages/Comments.jsx]: Just a small style nit, we generally omit parens on single arg functions

File: src/app/components/Comment/CommentReplyForm/styles.less [hunk 3]
```diff
@@ -27,7 +27,7 @@
     padding: 0.5em 0 0;
     float: left;
 
-    &[class*=" icon-"] {
+    &[class*=" icon-"], &[class*=" icon-"]:visited {
```
Comment (schwers) [inline src/app/components/Comment/CommentReplyForm/styles.less]: 🔕  This might be a holdover from 1X, do we still need a class selector here? All of our icons should have the `.icon` class in addition to the `icon-${iconName}` class

**Comment by schwers** [comment]
💅

**Comment by ajacksified** [comment]
:haircut_man:

File: src/app/components/Comment/CommentReplyForm/index.jsx [hunk 4]
```diff
@@ -4,22 +4,31 @@ import React from 'react';
...
 
-export const closeButton = (currentPage) => {
-  const href = urlFromPage(currentPage, { queryParams: { commentReply: undefined } });
+const T = React.PropTypes;
 
+export const closeButton = onToggleReply => {
   return (
-    <BackAnchor
-      href= { href }
+    <a
```
Comment (schwers) [inline src/app/components/Comment/CommentReplyForm/index.jsx]: Could this be a `<span />` or `<div />` instead ?

File: src/app/components/Comment/CommentTools/index.jsx [hunk 5]
```diff
@@ -63,17 +62,17 @@ CommentTools.defaultProps = {
...
 };
 
-const renderReply = ({ currentPage, id, commentReplying }) => {
-  const href = urlFromPage(currentPage, { queryParams: { commentReply: id } });
-
+const renderReply = ({ commentReplying, onToggleReply }) => {
   return (
-    <Anchor
-      href= { href }
+    <a
```
Comment (schwers) [inline src/app/components/Comment/CommentTools/index.jsx]: Same question on tag usage as above, I think having `<a />` makes this confusing. `onToggleReply` also now has increased complexity as it requires a call to `e.preventDefault()`, otherwise the page is going to scroll to the top when the browser looks for the `#` anchor

File: src/app/components/CommentsPage/CommentsPageTools/index.jsx [hunk 6]
```diff
@@ -69,9 +77,9 @@ export default ({ replying, post, hasSingleComment, currentPage, id, onSortChang
           onSortChange={ onSortChange }
         />
         { commentingDisabled ? null :
-          <Anchor className='Button m-linkbutton' href={ replyHref }>
+          <a className='Button m-linkbutton' href={ replyHref } onClick={ onToggleReply }>
```
Comment (schwers) [inline src/app/components/CommentsPage/CommentsPageTools/index.jsx]: Same `a` question

**Comment by schwers** [comment]
💅  Bunch of small nits on the usage of `<a />` tags, I think this needs to not use `<a />` at all to really remove the query param and its overhead. Could you re-look at the testing comment and let me know your thoughts?

**Comment by ajacksified** [comment]
:haircut_man:

File: src/app/pages/Comments.jsx [hunk 7]
```diff
@@ -64,11 +66,15 @@ const dispatchProps = dispatch => ({
   navigateToUrl(url, query) {
     dispatch(navigationActions.navigateToUrl(METHODS.GET, url, query));
   },
+  onToggleReply (e, id) {
+    e.preventDefault();
```
Comment (schwers) [inline src/app/pages/Comments.jsx]: I think `onToggleReply` is unneeded now right?

File: src/app/reducers/replying.js [hunk 8]
```diff
@@ -1,7 +1,7 @@
 import merge from '@r/platform/merge';
 import * as replyActions from 'app/actions/reply';
 
-const DEFAULT = {};
+export const DEFAULT = {};
```
Comment (schwers) [inline src/app/reducers/replying.js]: I don't see where this new export is being used

Comment (ajacksified) [inline src/app/reducers/replying.js]: Oh, this was cleanup from an existing file, not my new reducer - could be a separate commit instead if you'd like, or not part of this PR

File: src/app/components/Comment/index.jsx [hunk 9]
```diff
@@ -229,7 +230,7 @@ const selector = createSelector(
   (state, props) => state.comments[props.commentId],
   (state, props) => state.moreCommentsRequests[props.commentId] || DEFAULT_COMMENT_REQUEST,
   (state, props) => state.collapsedComments[props.commentId] || false,
-  (state, props) => state.platform.currentPage.queryParams.commentReply === props.commentId,
+  (state, props) => state.replyingComments[props.commentId] || false,
```
Comment (ajacksified) [inline src/app/components/Comment/index.jsx]: I was following the example from a line above - should I change everything instead?

**Comment by ajacksified** [comment]
:haircut_man:

File: src/app/reducers/replying.js (hunk 8 above)
Comment (schwers) [inline src/app/reducers/replying.js]: Okay, I think its fine to keep it in this patch.

**Comment by schwers** [comment]
Just a few changes (namely `!!` in selector, and exporting the default from your reducer for tests). :fish:'ing optimistically since those should be easy

**Comment by ajacksified** [comment]
A significant rebase, can you take a quick look, @schwers? :haircut_man:

File: src/app/actions/comment.js [hunk 10]
```diff
@@ -60,6 +60,16 @@ export const toggleCollapse = (id, collapse) => async (dispatch) => {
   dispatch(toggledCollapse(id, collapse));
 };
 
+export const TOGGLE_REPLY = 'COMMENT__TOGGLE_REPLY';
+
+export const toggledReply = (id, reply) => ({ type: TOGGLE_REPLY, id, reply });
+export const toggleReply = (id, reply) => async (dispatch) => {
```
Comment (nramadas) [inline src/app/actions/comment.js]: out of curiosity, why is this needed? it seems to just dispatch `toggledReply`. Couldn't whatever is dispatching this just dispatch `toggledReply` directly?

File: src/app/components/Comment/index.jsx [hunk 11]
```diff
@@ -269,8 +270,8 @@ const selector = createSelector(
   state => state.platform.currentPage,
   (state, props) => state.comments[props.commentId],
   (state, props) => state.moreCommentsRequests[props.commentId] || DEFAULT_COMMENT_REQUEST,
-  (state, props) => state.collapsedComments[props.commentId] || false,
-  (state, props) => state.platform.currentPage.queryParams.commentReply === props.commentId,
+  (state, props) => !!state.collapsedComments[props.commentId] || false,
```
Comment (nramadas) [inline src/app/components/Comment/index.jsx]: nit, but i think you don't have to `|| false` anymore since you're `!!`ing

File: src/app/actions/comment.js (hunk 10 above)
Comment (ajacksified) [inline src/app/actions/comment.js]: I was following the pattern set by collapse and some other toggle actions. It seemed extraneous, but I was going for consistency - happy to nuke it.

**Comment by ajacksified** [comment]
:haircut:

File: src/app/less/components/button.less [hunk 12]
```diff
@@ -1,8 +1,10 @@
-.Button, .Button:active, .Button:visited, .Button:hover, .Button:focus {
+.Button {
   padding: .5em 1em;
   letter-spacing: 1.6;
   line-height: 1.5;
+}
 
+.Button, .Button:active, .Button:visited, .Button:hover, .Button:focus {
```
Comment (schwers) [inline src/app/less/components/button.less]: 🔕  Small nit: this could be less typing if you nested it in the `.Button` selector above and use less's & shorthand for writing a new, non-nested rule. i.e. `&, &:active, &:visited, &:hover, &:focus`

**Comment by schwers** [comment]
:fish:

**Comment by ajacksified** [comment]
(merged @ 4a5bd54fa4ccbdaa2fbf90e88adc57dcac0c42f3)
